Objects/bytes_methods.c

```diff
@@ -557,10 +557,49 @@ find_internal(const char *str, Py_ssize_t len,
 }
 
 PyObject *
-_Py_bytes_find(const char *str, Py_ssize_t len, PyObject *sub,
+_Py_bytes_find(const char *str, Py_ssize_t len, PyObject *subobj,
                Py_ssize_t start, Py_ssize_t end)
 {
-    Py_ssize_t result = find_internal(str, len, "find", sub, start, end, +1);
+    Py_ssize_t result;
+    if (PyTuple_Check(subobj)) {
+        result = -1;
+        ADJUST_INDICES(start, end, len);
+        // Work in batches of 10000
+        for (; result == -1 && start <= end; start += 10000) {
+            for (Py_ssize_t i = 0; i < PyTuple_GET_SIZE(subobj); i++) {
+                PyObject *subseq = PyTuple_GET_ITEM(subobj, i);
+                Py_ssize_t sublen;
+                Py_buffer subbuf;
+                if (!PyObject_CheckBuffer(subseq)) {
+                    sublen = 1;
+                }
+                else if (PyObject_GetBuffer(subseq, &subbuf,
+                                            PyBUF_SIMPLE) != 0)
+                {
+                    return NULL;
+                }
+                else {
+                    sublen = subbuf.len;
+                }
+                Py_ssize_t cur_end = start + 10000 + sublen;
```

Comment:
```c
if (result > -1) {
    cur_end = start + sublen + result;
}
```

Reply:
Could be `cur_end = result + sublen`?
Where does `-1` come from? Is indexing different here or I am failing to follow something here?

Reply:
> Could be `cur_end = result + sublen`?

You are right.

> Where does -1 come from? Is indexing different here or am I failing to follow something here?

Think about `"foobar".find(("ob", "oba"))` and `"foobar".find(("ob", "oob"))`

"foobar".find("ob") => 2
cur_end = result + sublen -1 = 2 + 3 -1 = 4
"foob".find("oba") => -1  # its OK because result is 2 already
"foob".find("oob") => 1

Reply:
We can just use `cur_end = new_result - 1`, it's adjusted by `sub_len` anyway.